### DISCUSSION TITLE
Llamacpp inference with GPUS

### DISCUSSION THREAD

**Discussion by changye-chen** [oobabooga/text-generation-webui#2433] [Q&A] [answered]
Hey, I am trying command
```
python server.py --model guanaco-65B.ggmlv3.q4_0.bin --auto-devices --gpu-memory 22 22  --n-gpu-layers 80
```
on my 2*3090 device. but it seems can not offload layers to two gpus.

How could I do to solve this problem?

**Comment by BetaDoggo** [comment, marked as answer]
This isn't possible right now because it isn't supported by the [llama-cpp-python](https://github.com/abetlen/llama-cpp-python) library used by the webui for ggml inference. There's currently [a PR in the parent llama.cpp repo](https://github.com/ggerganov/llama.cpp/pull/1607) to refactor the cuda implementation which will make multi-gpu possible.

**Reply by changye-chen** [reply]
Got it. Thank you so much

**Comment by QuantumQuill** [comment]
does this mean, that I have to deactivate more then one GPU on an multi-GPU system? 
How can I do this?

**Reply by changye-chen** [reply]
No, it is mean that you can only use one gpu when using text-generation-webui. But you could use llama.cpp directly which BetaDoggo mentioned above

**Comment by evshiron** [comment]
GPTQ can be an alternative for now.
